metatheory/TestNEAT.hs

```diff
@@ -0,0 +1,105 @@
+module Main where
+
+import           Control.Monad.Except
+import           Data.Coolean
+import           Data.Either
+import           Language.PlutusCore
+import           Language.PlutusCore.Generators.NEAT.PropTest
+import           Language.PlutusCore.Normalize
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+import           MAlonzo.Code.Main                            (checkKindAgda, inferKindAgda, normalizeTypeAgda)
+import           MAlonzo.Code.Scoped                          (deBruijnifyK, unDeBruijnifyK)
+
+import           Language.PlutusCore.DeBruijn
+import           Raw
+
+
+main :: IO ()
+main = defaultMain allTests
+
+allTests :: TestTree
+allTests = testGroup "all tests"
+  [ testCaseCount "soundness" $
+      testTyProp depth kind prop_checkKindSound
+  , testCaseCount "normalization" $
+      testTyProp depth kind prop_normalizePreservesKind
+  , testCaseCount "normalizationSound" $
+      testTyProp depth kind prop_normalizeTypeSound
+  , testCaseCount "normalizationAgree" $
+      testTyProp depth kind prop_normalizeTypeSame
+  , testCaseCount "kindInferAgree" $
+      testTyProp depth kind prop_kindInferSame
+  ]
+
+testCaseCount :: String -> IO Integer -> TestTree
+testCaseCount name act = testCaseInfo name $
+  liftM (\i -> show i ++ " types generated") act
+
+
+-- NEAT settings
+
+depth :: Int
+depth = 10
```

Comment:
NEAT is size-based, not depth-based, right? In particular, this parameter is used as an argument to `ctrex'`, which is a generalized version of `ctrex`, which has the following docs:

> Find a counterexample to the given predicate, of or below a given size.

Reply:
Yes. Changed to `size`.